flax/core/frozen_dict.py

```diff
@@ -22,50 +22,42 @@
 K = TypeVar('K')
 V = TypeVar('V')
 
+
 @jax.tree_util.register_pytree_node_class
 class FrozenDict(Mapping[K, V]):
-  """An immutable variant of dictionaries.
-  """
-  __slots__ = ('_dict', '_leaves', '_treedef', '_hash')
+  """An immutable variant of the Python dict."""
+  __slots__ = ('_dict', '_hash')
 
   def __init__(self, *args, **kwargs):
-    if len(args) == 2:
-      self._treedef, self._leaves = args
-      self._dict = None
-    else:
-      self._dict = dict(*args, **kwargs)
-      self._treedef = None
-      self._leaves = None
+    xs = dict(*args, **kwargs)
+    self._dict = _freeze(xs)
     self._hash = None
     
   def __getitem__(self, key):
-    self._ensure_dict()
-    return self._dict[key]
+    v = self._dict[key]
+    if isinstance(v, dict):
```

Comment:
Is this necessary? Looks like _freeze already recursively freezes all subducts?

Reply:
I added a comment in _freeze